### ISSUE TITLE
Handling network errors (I/O timeout, connection timeout, etc)

### ISSUE THREAD

**Issue by virvum** [go-ldap/ldap#315]
I would like to handle network errors (such as I/O timeout, connection timeout, etc) separately. As far as I understand, this should be possible as follows:

```
if ldap.IsErrorWithCode(err, ldap.ErrorNetwork) {
    // handle network errors
} else if err != nil {
    // handle other errors
}
```
However, this doesn't seem to work. Errors like "ldap: connection timed out" still find their way into the `else if` branch.

What am I doing wrong?

**Comment by stefanmcshane** [comment]
Hi @virvum,

Thanks for catching this.
This looks like a bug/unintended feature caused [here](https://github.com/go-ldap/ldap/blob/c9551dc8a15abc270d5ea1a59366622cba6c09a5/v3/conn.go#L510)  as the error is created as `error.New` as opposed to using `NewError` as we do here.

Ill add a patch to this at the weekend and get it merged in. Alternatively, if you need it before then, feel free to open a PR, adjusting [these files/tests](https://github.com/go-ldap/ldap/search?q=connection+timed+out)

Thanks,
Stefan

**Comment by virvum** [comment]
Thank you very much for your feedback.

I created #316, please have a look.

**Comment by stefanmcshane** [comment]
Closing as merged with #316. Thanks for contributing @virvum
